### PR TITLE
Arsenal - Fill loadouts list over multiple frames

### PR DESCRIPTION
**Pull request by LinkIsGrim**
**When merged this pull request will:**
- #9785 but for loadouts

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/arsenal/functions/fnc_fillLoadoutsList.sqf [hunk 1]
```diff
@@ -1,42 +1,60 @@
 #include "..\script_component.hpp"
 #include "..\defines.hpp"
 /*
- * Author: Alganthe, johnb43
+ * Author: Alganthe, johnb43, LinkIsGrim
  * Fill loadouts list.
```
Comment (johnb432) [inline addons/arsenal/functions/fnc_fillLoadoutsList.sqf:5]: Maybe add something in the description about the function filling 10 loadouts per frame?

File: addons/arsenal/functions/fnc_fillLoadoutsList.sqf [hunk 2]
```diff
@@ -1,42 +1,60 @@
...
 private _newRow = -1;
 
 if (GVAR(currentLoadoutsTab) != IDC_buttonSharedLoadouts) then {
     private _loadoutNameAndTab = "";
     private _loadoutCachedInfo = "";
     private _sharingEnabled = GVAR(allowSharedLoadouts) && {isMultiplayer};
+    private _loadouts = [
+        profileNamespace getVariable [QGVAR(saved_loadouts), []],
+        GVAR(defaultLoadoutsList)
+    ] select (ctrlIDC _control == IDC_buttonDefaultLoadouts);
```
Comment (johnb432) [inline addons/arsenal/functions/fnc_fillLoadoutsList.sqf:52]: Isn't there a chance that these variables could change over the course of loading?

Comment (LinkIsGrim) [reply addons/arsenal/functions/fnc_fillLoadoutsList.sqf:52]: Contents could change, menu would need to be refilled for changes to take effect

Comment (johnb432) [reply addons/arsenal/functions/fnc_fillLoadoutsList.sqf:52]: My question is: Can it break due to that?

Comment (LinkIsGrim) [reply addons/arsenal/functions/fnc_fillLoadoutsList.sqf:52]: loadouts with index divisible by `LOADOUTS_PER_FRAME` + number of deleted loadouts (so the index 10 until 10 + n, 20->20 +n, 30->30 +n... loadouts, where n is how many loadouts with lower indexes were removed from the array) could potentially be skipped if a loadout from an earlier batch was removed from the loadout array before next frame.

Loadouts that have their contents altered would keep their contents at whatever frame they were processed (loadout L1 is added at frame 0, get its contents changed before frame 1, list is filled during frame 1, if a player tries to load L1 they'll load the content it had during frame 0.

Due to the syntax of select used the only hard break is nil-ing the loadout array entirely.

Comment (johnb432) [reply addons/arsenal/functions/fnc_fillLoadoutsList.sqf:52]: What about multiple function calls, one every frame?

Comment (LinkIsGrim) [reply addons/arsenal/functions/fnc_fillLoadoutsList.sqf:52]: only way for that to happen is for the display to be opened & closed multiple times, at which point the loop will abort on a null display

**Review by johnb432 — APPROVED** [review]

**Pull request merged**